controllers/pipeline/pipeline_adapter_test.go

```diff
@@ -462,6 +462,27 @@ var _ = Describe("Pipeline Adapter", Ordered, func() {
 		}, time.Second*10).Should(BeTrue())
 	})
 
+	When("Snapshot already exists", func() {
+		BeforeEach(func() {
+			adapter = NewAdapter(testpipelineRunBuild, hasComp, hasApp, ctrl.Log, k8sClient, ctx)
+			Expect(reflect.TypeOf(adapter)).To(Equal(reflect.TypeOf(&Adapter{})))
+
+			snapshot, err := adapter.prepareSnapshotForPipelineRun(adapter.pipelineRun, adapter.component, adapter.application)
+			Expect(err).To(BeNil())
+
+			err = adapter.client.Create(adapter.context, snapshot)
+			Expect(err).To(BeNil())
+		})
+
+		It("ensures snapshot creation is skipped when snapshot already exists", func() {
+			Eventually(func() bool {
+				adapter.EnsureSnapshotExists()
```

Comment:
I don't understand why this `adapter.EnsureSnapshotExists()` is called twice, could you explain? Thanks!

Reply:
my bad, removed. Nice catch!

Reply:
Thanks!